### ISSUE TITLE
Anomalies building RACHEL images

### ISSUE THREAD

**Issue by mannyack** [learningequality/ka-lite#3559]
I am in the process of rebuilding the RACHEL images for worldpossible.org that will be compatible with the older models and the new RPi2. As you may know, KA-Lite is included in the 64gb version of RACHEL.

I successfully installed KA-Lite, following the instructions on your web-site for installation on Linux, not the extended instructions for the RPi wifi. Because of the issues listed below, I tried the installation on a model B (not B+) and a model 2. In both cases the installation appeared to be successful and I was able to access KA-Lite, given the anomalies below. Since I used git-clone without specifying a branch, I assume that I installed the master branch.

After installation I noticed two anomalies. I tried moving the image that was created on the model B to a micro SD card and run it on the model 2 and vice versa. I also tried both images on a model B+. The anomalies were identical in all cases.

1) On the B and B+, KA-Lite was accessible on port 7007 but not 8008. On the model 2, the opposite was observed, i.e. KA-Lite was accessible on port 8008, but not on 7007. Expected behavior would have had them available on the same ports for all RPi models.

2) I thought that perhaps optimizerpi.sh did not run successfully, so I reran it manually. It generated an error message that in step two, it was looking for manage.py in ka-lite/scripts, not where the file actually is, ka-lite/kalite. I tried hard coding the correct path in optimizerpi.sh for manage.py, and it was great for the B and B+, that is, KA-Lite is now available on both port 7007 and 8008. However when I tried to boot the model 2 with this image, there were error messages. here is a snippet of the boot crawl including the error messages.

Starting ka-lite!

Trying to start the web server.
INFO:kalite:Invalidating the web cache.
INFO:kalite:Running videoscan.
INFO:kalite:Preloading topic data.

To access KA Lite from another connected computer, try the following address(es):
    http://10.10.10.10:8008/
    http://127.0.0.1:8008/
    http://192.168.1.141:8008/
WARNING:root:Port 8008 is busy
Exception: Port 8008 is currently in use by another process, cannot continue
Error: The web server was not started
[ ok ] Starting MySQL database server: mysqld . ..
[info] Checking for tables which need an upgrade, are corrupt or were
not closed cleanly..
My IP address is 192.168.1.141 10.10.10.10
[FAIL] startpar: services(s) returned failure: kalite ... failed!

Raspbian GNU/Linux 7 rachel tty1

rachel login:

I have a resolution (start-up shell script that sets the port in our html based on the contents of /proc/cpuinfo), but I am still looking for a better one.

And by the way, let's talk about the message suggesting that we try to access KA Lite from "another connected computer" on 127.0.0.1:8008. Or rather, let's not.

**Comment by jamalex** [comment]
Version 0.13 will be released on Monday (after 10 months in the making), and would be the best version to use for this. A number of the issues (e.g. [7007 vs 8008](https://github.com/learningequality/ka-lite/issues/2695)) have been resolved in this new version, though we haven't been able to do testing with a Model 2 yet (would be great to hear how it goes).

The new version lives on the "0.13.x" branch. As part of the release on Monday, we will be merging that branch to master, but if you want to test prior to that, you can try the instructions here:
https://groups.google.com/a/learningequality.org/forum/#!topic/dev/ve6RxN647ZA

(you may wish to [subscribe to the dev mailing list](https://groups.google.com/a/learningequality.org/forum/#!forum/dev) to stay more in the loop, if you haven't already)

**Comment by mannyack** [comment]
Hi Jamie,

What I have done testing your new version so far:

1) Installed 0.13 on RPi model 2.
- Started with the basic RACHEL image I am building which is based on 
the raspbian image from 2015-02-16.
- Deleted other Khan modules.
- git clone for the 0.13 version
- copied assessment.zip into /home/pi
- bin/kalite manage setup
- copied a large set of videos into ka-lite/content
- started ka-lite
- rebooted
- started ka-lite
- logged in as admin from Firefox running on my Windows desktop.
-- completed registration
-- looked at the topic tree under the manage->videos
-- downloaded english subtitles
-- watched two videos
-- did some exercises on two pages

Observations:
-- KA-Lite is running on port 8008, not on 7007.
-- New interface is interesting and will take some time to get used to.
-- Everything looked good, except I did not see how to get the subtitles 
to play.
-- The model 2, here are the message in the message crawl after 
bin/kalite start

INFO:kalite:Invalidating the web cache.
INFO:kalite:Preloading exercise data for language en.
INFO:kalite:Preloading content data for language en.
INFO:kalite:Preloading topic tree data for language en.

0 static files copied, 2396 unmodified.
To access KA Lite from another connected computer, try the following 
address(es):
     http://10.10.10.10:8008/
     http://127.0.0.1:8008/
     http://192.168.1.143:8008/

(The 10.10.10.10 address is because the model 2 has a configured wifi 
usb dongle.)

2) Installed 0.13 on RPi B.
- Same as the model 2 installation, except that I copied the videos into 
ka-lite/content after the git-clone and before the bin/kalite manage setup.

Observations:
-- KA-Lite is running on port 7007, not on 8008
-- Same new interface.
-- Same problem with the subtitles.
-- On the model B, the message crawl after bin/kalite start was:

INFO:kalite:RPi package selected.
INFO:kalite:Invalidating the web cache.
INFO:kalite:Preloading exercise data for language en.
INFO:kalite:Preloading content data for language en.
INFO:kalite:Preloading topic tree data for language en.

0 static files copied, 2396 unmodified.
To access KA Lite from another connected computer, try the following 
address(es):
     http://127.0.0.1:8008/
     http://192.168.1.143:8008/

(The model B does not have a configured wifi usb dongle.)

Comparison:
-- It is very easy to observe the speed difference between the two 
machines on the server (during build and configuration) and the client 
(when running KA-Lite).
-- Again...the ports are different.
-- On the model B, kalite appears to have realized it was running on a 
RPi; on the model 2, it did not. (Look at first message of the message 
crawl in both cases.)

Manny

PS You can send email directly to me at manny@worldpossible.org or 
mannyack@mail.com (No, I did not misspell gmail, I did mean mail.)

**Comment by MCGallaspy** [comment]
Thanks for your detailed observations @mannyack. Just to be clear, you're saying in some circumstances you expect the `bin/kalite start` command to serve from a different port? Also, noted that:
> To access KA Lite from another connected computer, try the following
address(es):
     http://127.0.0.1:8008/

Is probably not very helpful.

**Comment by mannyack** [comment]
I think I gave you all the details earlier. If anything is not clear, 
let me know and I can pass along documentation.

You did not quite say the first part correctly. KA-Lite does server from 
different ports, depending on the physical server. I think there should 
be no difference between the port served when the server is any type of 
Raspberry Pi, which is not the case right now.

Manny

**Comment by benjaoming** [comment]
It's true there used to be a conditional statement in the settings where PRODUCTION_PORT was 7007 if the "RPi" config package was selected.. and PROXY_PORT was set to 8008, I have no idea what that actually means.

But this seems somewhat redundant now:

 1. We do not use django settings for the port, we use `kalite start --port=8008` #3827 
 1. The notion of config packages are being deprecated in favor of a default RPi settings module `--settings=kalite.project.settings.raspberry_pi`.

Here's the horror code from the old settings

```python
# Config for Raspberry Pi distributed server
if package_selected("RPi"):
    LOG.info("RPi package selected.")
    # nginx proxy will normally be on 8008 and production port on 7007
    # If ports are overridden in local_settings, run the optimizerpi script
    PRODUCTION_PORT = getattr(local_settings, "PRODUCTION_PORT", 7007)
    PROXY_PORT = getattr(local_settings, "PROXY_PORT", 8008)
    assert PRODUCTION_PORT != PROXY_PORT, "PRODUCTION_PORT and PROXY_PORT must not be the same"
    #SYNCING_THROTTLE_WAIT_TIME = getattr(local_settings, "SYNCING_THROTTLE_WAIT_TIME", 1.0)
    #SYNCING_MAX_RECORDS_PER_REQUEST = getattr(local_settings, "SYNCING_MAX_RECORDS_PER_REQUEST", 10)

    PASSWORD_ITERATIONS_TEACHER = getattr(local_settings, "PASSWORD_ITERATIONS_TEACHER", 2000)
    PASSWORD_ITERATIONS_STUDENT = getattr(local_settings, "PASSWORD_ITERATIONS_STUDENT", 500)

    ENABLE_CLOCK_SET = getattr(local_settings, "ENABLE_CLOCK_SET", True)

    DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP = True
```

**Comment by aronasorman** [comment]
Hi @mannyack, we've released 0.14 with a special deb file for the RPi. Please see the instructions [here](https://github.com/learningequality/ka-lite/pull/3932) and open another issue if you find issues with that.
